### PR TITLE
adds /coverageIgnore option to exclude files from code coverage results

### PR DESCRIPTION
**Pull request by stogle**
My implementation of a solution for #404. Adds a /coverageIgnore command-line option (and corresponding "CodeCoverageIgnores" option in chutzpah.json), much like the existing /coverageInclude and /coverageExclude options. Ignored files are not added to the CoverageData object, thus don't appear in the coverage output.

### CODE REVIEW

File: Chutzpah/Coverage/BlanketJsCoverageEngine.cs [hunk 1]
```diff
@@ -166,7 +168,7 @@ public CoverageData DeserializeCoverageObject(string json, TestContext testConte
                     }
                 }
 
-                if (IsFileEligibleForInstrumentation(newKey) && fileSystem.FileExists(filePath))
```
Comment (mmanela) [inline Chutzpah/Coverage/BlanketJsCoverageEngine.cs]: Why did you remove the check on IsFileEligibleForInstrumentation?

Comment (stogle) [inline Chutzpah/Coverage/BlanketJsCoverageEngine.cs]: That seems like an error on my part.

Comment (mmanela) [inline Chutzpah/Coverage/BlanketJsCoverageEngine.cs]: Please fix and validate then I can merge

**Comment by stogle** [comment]
I have restored the check for IsFileEligibleForInstrumentation.

**Comment by mmanela** [comment]
Cool thanks!

**Comment by mmanela** [comment]
@stogle I merged the changed but I forgot to check the functional test. It looks like the ones you added are failing. For example:

```
Chutzpah.Facts.Integration.Coverage.Will_ignore_only_given_file_patterns(scriptPath: "JS\Code\RequireJS\MochaWithSettings\all.tests.moch"...) [FAIL]
   Assert.Equal() Failure
   Expected: \base\core.js, ui\screen.js
   Actual:   C:\Dev\chutzpah\Facts.Integration\bin\Debug\JS\Code\RequireJS\tests\ui\ui.mocha-qunit.test.js, C:\Dev\chutzpah\Facts.Integration\bin\Debug\JS\Code\RequireJS\ui\screen.js, C:\Dev\chutzpah\Facts.Integration\bin\Debug\JS\Code\RequireJS\base\core.js
   Stack Trace:
      c:\Dev\chutzpah\Facts.Integration\Coverage.cs(518,0): at Chutzpah.Facts.Integration.Coverage.ExpectKeysMatching[T](IDictionary`2 dict, IEnumerable`1 keySubstrings)
      c:\Dev\chutzpah\Facts.Integration\Coverage.cs(449,0): at Chutzpah.Facts.Integration.Coverage.Will_ignore_only_given_file_patterns(String scriptPath)
```
